src/tests/multihost/alltests/test_inmemory_cache.py

```diff
@@ -0,0 +1,329 @@
+        client.run_command("service sssd stop")
+        client.run_command("sss_cache -U --domain=example1")
+        client.run_command("sss_cache -G --domain=example1")
+        client.run_command("service sssd start")
+        clear_only_domain_log(multihost)
+        client.run_command("getent passwd foo1")
+        client.run_command("getent group ldapusers")
+        find_logs(multihost, log_sssd, "Got request for ")
+        find_logs(multihost, log_sssd, "uid=foo1")
+        find_logs(multihost, log_sssd, "cn=ldapusers")
+
+    @staticmethod
+    def test_memcache(multihost):
+        """
+        :title: Set value of option memcache timeout to 0 and expire InMemory cache immediately
+        :id: 19dfa7ae-96e9-11ed-adf3-845cf3eff344
+        :steps:
+            1. Set value of option memcache timeout to 0 and expire InMemory cache immediately
+            2. Validate the user using sss cache
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+        """
+        client = multihost.client[0]
+        tools = sssdTools(multihost.client[0])
+        tools.sssd_conf("nss", {'memcache_timeout': '0'}, action='update')
+        tools.clear_sssd_cache()
+        client.run_command("strace -o /tmp/stracefoo1 getent passwd foo1")
+        find_logs(multihost, "/tmp/stracefoo1", "/var/lib/sss/pipes/nss")
+        time.sleep(2)
+        client.run_command("strace -o /tmp/stracefoo1 getent passwd foo1")
+        find_logs(multihost, "/tmp/stracefoo1", "/var/lib/sss/pipes/nss")
+        time.sleep(5)
+        client.run_command("strace -o /tmp/stracefoo1 getent passwd foo1")
+        find_logs(multihost, "/tmp/stracefoo1", "/var/lib/sss/pipes/nss")
+        time.sleep(5)
+        client.run_command("strace -o /tmp/stracefoo1 getent passwd foo1")
+        find_logs(multihost, "/tmp/stracefoo1", "/var/lib/sss/pipes/nss")
+
+    @staticmethod
+    def test_bz1484376(multihost):
+        """
+        :title: Add a configuration option to SSSD to disable the memory cache bz1484376
+                Set value of option Memcache timeout to 35
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1484376
+        :id: b64237b4-96ef-11ed-adb1-845cf3eff344
+        :steps:
+            1. Add a configuration option to SSSD to disable the memory cache
+            2. Set memcache_timeout=0
+            3. Set value of option Memcache timeout to 35
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+            3. Should succeed
+        """
+        client = multihost.client[0]
+        client.run_command("rm -rvf /var/lib/sss/mc/*")
+        client.run_command("systemctl restart sssd")
+        # MC directory is empty, result as expected.
+        assert client.run_command("ls -A /var/lib/sss/mc/").stdout_text == ''
+        tools = sssdTools(multihost.client[0])
+        tools.sssd_conf("nss", {'memcache_timeout': '35'}, action='update')
```

Comment:
You are changing the config, backupsssdconf fixture should be used to restore the config.